### PR TITLE
uiua: init at unstable-2023-09-28

### PR DESCRIPTION
**Pull request by cafkafk**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cafkafk** [comment]
This is a bit ugly I'll clean it up in asap

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 257673` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>uiua</li>
  </ul>
</details>

**Comment by cafkafk** [comment]
This is ready for review.

File: pkgs/by-name/ui/uiua/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchFromGitHub {
+    owner = "uiua-lang";
+    repo = "uiua";
+    rev = "f5cd0bc4f1f2e34c8b75324c23e9b2db3f1c6727";
+    hash = "sha256-Os9soRokxxlh/+Jd5L0K/NWxiSBc8ILe6MAA92W95LI=";
+  };
+
+  # https://github.com/uiua-lang/uiua/issues/8
+  cargoLock.lockFile = ./Cargo.lock;
```
Comment (wackbyte) [inline pkgs/by-name/ui/uiua/package.nix:20]: A `Cargo.lock` file was added in a recent commit, so I would recommend updating.

Comment (cafkafk) [reply pkgs/by-name/ui/uiua/package.nix:20]: Ohh, well, here I go, one moment.

File: pkgs/by-name/ui/uiua/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  doCheck = false;
```
Comment (TomaSajt) [inline pkgs/by-name/ui/uiua/package.nix:25]: Tests ran fine for me, is there a reason why this is disabled?

File: pkgs/by-name/ui/uiua/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  pkgs,
+  stdenv,
```
Comment (TomaSajt) [inline pkgs/by-name/ui/uiua/package.nix:4]: `pkgs` and `stdenv` don't seem to be used

File: pkgs/by-name/ui/uiua/package.nix (hunk 2 above)
Comment (cafkafk) [reply pkgs/by-name/ui/uiua/package.nix:21]: They were unreasonably slow for me, but I'll take another look.

File: pkgs/by-name/ui/uiua/package.nix (hunk 1 above)
Comment (cafkafk) [reply pkgs/by-name/ui/uiua/package.nix]: Alright, it should be removed now, and it did seem to still build fine.

File: pkgs/by-name/ui/uiua/package.nix (hunk 2 above)
Comment (cafkafk) [reply pkgs/by-name/ui/uiua/package.nix]: I've enabled them. I'm running them locally now, but thought I'd give ofborg a headstart.

**Comment by TomaSajt** [comment]
I see you bumped the version, could you also show this in the commit message and PR title?

**Comment by cafkafk** [comment]
> I see you bumped the version, could you also show this in the commit message and PR title?

Yes, ofc, done!

File: pkgs/by-name/ui/uiua/package.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage rec {
```
Comment (wegank) [inline pkgs/by-name/ui/uiua/package.nix:6]:
```suggestion
rustPlatform.buildRustPackage {
```

**Comment by cafkafk** [comment]
> I did some testing, and it turns out you only need alsa-lib in buildInputs

Ahh nice, thanks, then we can throw out `udev`.

> Also, I don't really know what the right name for a flag like this should be. The linked derivation used `enableSound` but I think `enableAudio` or `audioSupport` would also be correct.

I personally prefer `audioSupport` over `enable sound`. The term `enable sound` seems like it automatically produces sound, while `audioSupport` suggests it provides the capability to produce sound.

File: pkgs/by-name/ui/uiua/package.nix (hunk 4 above)
Comment (cafkafk) [reply pkgs/by-name/ui/uiua/package.nix]: Ahh yes, fixed it, thanks!

**Comment by cafkafk** [comment]
Comments are addressed now

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 257673` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>uiua</li>
  </ul>
</details>

**Pull request merged**